### DISCUSSION TITLE
List instruments in .syx files

### DISCUSSION THREAD

**Discussion by DutchBigEd** [probonopd/MiniDexed#544] [Ideas]
I've programmed a .vi in labview that shows all the instruments in .syx files within a folder.
But strange things happen with some (not all) .syx files. Look:
in the 00177_DX7-YS200-1.syx file it gives:
Elegant   
SoftBrass 
WideString
Cosmic    
LargePipes
SynString1
FolkGuitar
Piano 1   
E.Piano 1 
DistGuitar
SoftString
SynString2
RichString
SynBrass 1
SynBrass 2
SynBrass 3
BrethBrass
SoftEns.  
WarmEns.  
OrchesEns.
Sunbeam   
Shimmer 1 
SoftCloud 
Bamarimba 
Sandarimba
FloatChime
Daybreak  
Tinqule   
SandBell  
Suspense  
Fog       
HuskyVoice

But in the 00218_Tfi1.syx file (to just name one), it shows:
AC.PNO 1.1
STRGS  2.1
.ORG  3.1@
OICE  4.1*
RMPT  5.1_
.PNO  6.1T
ELLO  7.1c
.ORG  8.1=
.HORN 9.1=
CSYN 10.1K
MPNO 11.1c
RIGL 12.1c
.PNO 13.1_
.PNO 14.1T
TGSM 15.1c
LUK  16.1_
PNO 17.1c<
TBS 18.1#X
NGL 19.1U
SYN 20.15@
RGS 21.1M8
BNE 22.1TM
LLO 23.1cM
MAL 24.1$
ICE 25.1cc
CLD 26.1IE
.WC.27.1_H
RPS 28.1bW
AV. 29.1c0
BES 30.1`Y
.HP.31.1c
TRE 32.1&÷

On the Oled display of the Minidexed it shows the correct name...
puzzled...

**Comment by probonopd** [comment]
Packed vs. not packed maybe?

* https://github.com/rarepixel/dxconvert/blob/master/FORMATS.txt
* Unpacking code in Python: https://github.com/bwhitman/learnfm/blob/d7bc524c414ec06b8bdf5b7bfedcba2c38a4e0b3/dx7db.py#L78

**Reply by DutchBigEd** [reply]
Thanks, but no...
I found my error, I was reading it as a text file, but had to read it as a binary file, because since it is data, there can (and will be) unprintable characters in these files, so counting characters will sometimes mis one due to being unprintable of the character.

See:

00218_Tfi1.syx
AC.PNO 1.1
STRGS  2.1
P.ORG  3.1
VOICE  4.1
TRMPT  5.1
E.PNO  6.1
CELLO  7.1
E.ORG  8.1
E.HORN 9.1
PCSYN 10.1
FMPNO 11.1
TRIGL 12.1
E.PNO 13.1
E.PNO 14.1
STGSM 15.1
PLUK  16.1
A.PNO 17.1
B.TBS 18.1
DINGL 19.1
P.SYN 20.1
STRGS 21.1
TRBNE 22.1
CELLO 23.1
AFMAL 24.1
P.ICE 25.1
FLCLD 26.1
GL.WC.27.1
HARPS 28.1
CLAV. 29.1
VIBES 30.1
DB.HP.31.1
BLTRE 32.1

**Comment by Banana71** [comment]
Hi Ed,

I'll try to describe briefly how I created the sound list. My spreadsheet program is set to German but I think you can get by with it. In the zip file you will find everything you need. 
If I upload new performances in the future I will include the sound list. (free service)
[Manual Soundliste.zip](https://github.com/probonopd/MiniDexed/files/12448226/Manual.Soundliste.zip)


With Windows 11 I open the miniDexed SD card and go to the folder with the performanes. There I mark all performances and press CTRL+Shift+C. (copy all paths)
![Pfade kopieren](https://github.com/probonopd/MiniDexed/assets/104296205/3a791676-6ad4-4599-b5fb-414b989cb32c)

Then I open the table SoundListe.ods (spreadsheet, LibreOffice, Excel,...) and insert the paths with SRTG+V. I import the paths with the following options. 
![STRG-V import in Tabelle](https://github.com/probonopd/MiniDexed/assets/104296205/92debe5c-63cf-45d4-a51b-6b1148c26bde)

You can see the corresponding formulas in the pictures or in the table. Then all you have to do is bring the table into the desired format, done. :-)
![Tab Formeln](https://github.com/probonopd/MiniDexed/assets/104296205/75ffaa06-8942-44e5-a036-d15344f38c3c)

**Reply by DutchBigEd** [reply]
Thank you for the effort, but I’m terrible with spreadsheet programs.
I thought that the list shown was about instruments in a .syx file, interesting because they are not shown in explorer.
For performances I just make a screenshot of the list of files in the performance folder. For me that suffices.
But again, thank you for the effort, and maybe someone else can benefit from it 😃👍🏻

**Comment by probonopd** [comment]
Should we script this?
